### PR TITLE
refactor: Repository

### LINKED ISSUE #17: Refactor repository
Remove the EmptyBot project and expose only the runtime solution code, including refactoring the existing code to their own namespaces.

### PR DESCRIPTION
**Pull request by peterinnesmsft**
## Description
Close #17 

Performing some refactoring of the botframework-core repo to help position the various different project points for further iteration and investment from the team.

## Specific Changes
  - Removing Composer-specific elements from the repo (except for Custom Actions, which needs to ultimately be incorporated into the main WebHost application).
  - Raising the runtime solution up to the root of the repo directory, and renaming to BotRuntime.sln. Renaming projects to `Microsoft.Core.Runtime`, `Microsoft.Bot.Runtime.WebHost`, `Microsoft.Bot.Runtime.CustomAction` and `Microsoft.Bot.Core.Tests` respectively, and removing Composer-specific naming from some of these elements.
  - Applying the Startup refactoring changes that Tracy has incorporated so far.

## Testing
The majority of the unit tests were testing dialogs that are included by out-of-the-box projects in Composer. I have removed those since they are not applicable in the scope of this repo. For the SkillConversationIdFactoryTests, I have refactored them into a single unit test based on the contents of the existing one and tidied up the code in general.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,360 @@
-﻿################################################################################
-# This .gitignore file was automatically created by Microsoft(R) Visual Studio.
-################################################################################
+## Ignore Visual Studio temporary files, build results, and
```
Comment (peterinnesmsft) [inline .gitignore:1]: The contents of this .gitignore file are simply copied from the inner one that existed for the WebApp project.

**Review by ryanlengel — DISMISSED** [review]
This mostly aligns with what I was thinking, nice!

Two comments/questions:
* I was thinking Custom Actions would be under `/src` like the other projects and stripped of Composer naming. Any particular reason why that's left out?
* Naming: Runtime makes sense to me as the entire solution (web host + library + tests), but for the library itself `Microsoft.Bot.Runtime` it seems confusing. i.e. in the future if I create a bot called `MyBot` and eject the runtime, I get a web host application plugged into the runtime library? Perhaps this is where we should include the Core terminology.

**Comment by tracyboehrer** [comment]
> * Naming: Runtime makes sense to me as the entire solution (web host + library + tests), but for the library itself `Microsoft.Bot.Runtime` it seems confusing. i.e. in the future if I create a bot called `MyBot` and eject the runtime, I get a web host application plugged into the runtime library? Perhaps this is where we should include the Core terminology.

Is there another word for "Core"?  That's another word that's tossed around a lot.  Also, people often say "Core Runtime".  i.e., "Core Runtime" is "Runtime" which uses "Core".  Hmmm.

**Comment by ryanlengel** [comment]
Based on offline conversations I've made some adjustments to the naming and structure.
image

Validated by testing against the @microsoft/botframework-virtual-assistant repository. To test the same yourself
1. Copy this source code into that repository under /Assistant/runtime/
1. In Composer, change the start command to `dotnet run --project src/Microsoft.Bot.Runtime.WebHost`

image

I've also added a variety of #TODO comments and included a bot on the repository that should convert these into issues when the PR is merged in (assigning to the owner of the commit).

File: src/Microsoft.Bot.Core/LuisConfig.cs [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 using System.Collections.Generic;
 
-namespace Microsoft.BotFramework.Composer.Core
+namespace Microsoft.Bot.Runtime
```
Comment (peterinnesmsft) [inline src/Microsoft.Bot.Core/LuisConfig.cs:6]: Change namespace to Microsoft.Bot.Core.

File: src/Microsoft.Bot.Core/Microsoft.Bot.Core.csproj [hunk 3]
```diff
@@ -5,11 +5,11 @@
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
-    <CodeAnalysisRuleSet>Microsoft.BotFramework.Composer.Core.ruleset</CodeAnalysisRuleSet>
+    <CodeAnalysisRuleSet>Microsoft.Bot.Runtime.ruleset</CodeAnalysisRuleSet>
```
Comment (peterinnesmsft) [inline src/Microsoft.Bot.Core/Microsoft.Bot.Core.csproj:8]: Should this now be Microsoft.Bot.Core.ruleset?

File: src/Microsoft.Bot.Runtime.CustomAction/Microsoft.BotFramework.Composer.CustomAction.csproj [hunk 4]
```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (peterinnesmsft) [inline src/Microsoft.Bot.Runtime.CustomAction/Microsoft.BotFramework.Composer.CustomAction.csproj:1]: Do we still need this file?

**Review by peterinnesmsft — COMMENTED** [review]
:shipit:

File: src/Microsoft.Bot.Runtime.CustomAction/Microsoft.BotFramework.Composer.CustomAction.csproj (hunk 4 above)
Comment (ryanlengel) [reply src/Microsoft.Bot.Runtime.CustomAction/Microsoft.BotFramework.Composer.CustomAction.csproj]: Removed

**Review by ryanlengel — APPROVED** [review]

**Pull request merged**